packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -520,6 +531,13 @@ class _ButtonStyleState extends State<ButtonStyleButton> with TickerProviderStat
         minSize = Size.zero;
     }
 
+    if (widget.tooltip != null) {
+      result = Tooltip(
+        message: widget.tooltip,
+        child: result,
+      );
+    }
```

Comment:
The G3 failure helps catch a corner case, with the current change(`Tooltip` is parent of `InkWell`), the extra tap area(the black area) doesn't work anymore.

<details><summary>Sample code</summary>

```dart

import 'package:flutter/material.dart';

/// Flutter code sample for [Dialog].

void main() => runApp(const DialogExampleApp());

class DialogExampleApp extends StatelessWidget {
  const DialogExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Scaffold(
        body: DialogExample(),
      ),
    );
  }
}

class DialogExample extends StatelessWidget {
  const DialogExample({super.key});

  @override
  Widget build(BuildContext context) {
    return const Center(
      child: ColoredBox(
        color: Colors.black,
        child: BackButton(
          color: Colors.white,
          style: ButtonStyle(
            padding: WidgetStatePropertyAll<EdgeInsets>(EdgeInsets.all(20)),
            backgroundColor: WidgetStatePropertyAll<Color>(Colors.yellow)
          ),
        ),
      ),
    );
  }
}

```

</details>

<details><summary>Demo</summary>

https://github.com/user-attachments/assets/d20568b2-3a3a-44d8-9ec1-ffcc9d77689b

</details>

If we put tooltip inside of `InkWell`, tapping area works as expected. It's probably because of the way how tooltip treats its parent widget(I haven't dig too deep into this:)).

<details><summary>After fix</summary>

https://github.com/user-attachments/assets/e73f817a-126d-40de-ba2e-40845bfd5b3a

</details>

Reply:
Thank you for taking a look! I updated the fix and added another test to catch this exact scenario within icon_button_test class

Reply:
Sounds good! Feel free to land it whenever you are ready!